AirParrot: Add uninstall/zap

MindTooth opened the pull request:
<!-- If there’s a checkbox you can’t complete for any reason, that's okay, just explain in detail why you weren’t able to do so. -->

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

commitay commented on Casks/airparrot.rb:
```diff
@@ -9,4 +9,19 @@
   app "AirParrot #{version.major}.app"
+
+  uninstall kext:   [
+                      '/Library/Extensions/AirParrotDriver.kext',
```
Shouldn't need to pass the full path, just the kext id. `uninstall kext:` should also delete the kext.

MindTooth replied:
Thanks.  I'll shorten the name, however the two `SharedExtensions` did not delete :(  I needed to force them.  Not sure why really  😕

MindTooth replied:
@commitay:  Afraid to push changes in case you don't see my comment 😨

adidalal replied:
@MindTooth Did you mean `StagedExtensions`?

Can you try looking for them with `/usr/sbin/kextstat -l -b {kext_name}` and see what comes up?

(as found in https://github.com/Homebrew/brew/blob/af9ddfe899c3593a4d8168d4ade61eccd01d19b0/Library/Homebrew/cask/lib/hbc/artifact/abstract_uninstall.rb#L167)

MindTooth commented:
I ment `StagedExtensions` 😄 

`/usr/sbin/kextstat -l -b AirParrotDriver` is empty.

Trimmed it down to:
```ruby
  uninstall kext: 'AirParrotDriver.kext'
```

But all four is still present on my system:
```shell
➜ sudo fd 'airparrot' /
/Library/Extensions/AirParrotDriver.kext
/System/Library/Extensions/AirParrotDriver.kext
/Library/StagedExtensions/Library/Extensions/AirParrotDriver.kext
/Users/mt/Library/Caches/com.squirrels.AirParrot-2
/Library/StagedExtensions/System/Library/Extensions/AirParrotDriver.kext
/usr/local/var/homebrew/locks/airparrot--2.7.3.dmg.incomplete.lock
/Library/Extensions/AirParrotDriver.kext/Contents/MacOS/AirParrotDriver
/private/var/folders/nk/qf5y0f053kb_vrxyjdgvsynh0000gn/C/com.squirrels.AirParrot-2
/System/Library/Extensions/AirParrotDriver.kext/Contents/MacOS/AirParrotDriver
/Users/mt/Library/Caches/Homebrew/Cask/airparrot--2.7.3.dmg
/Library/StagedExtensions/Library/Extensions/AirParrotDriver.kext/Contents/MacOS/AirParrotDriver
/Library/StagedExtensions/System/Library/Extensions/AirParrotDriver.kext/Contents/MacOS/AirParrotDriver
```

Meaning I still have the device under Sound Output:
<img width="780" alt="screen shot 2018-06-09 at 11 50 36" src="https://user-images.githubusercontent.com/33870508/41190249-55b7b1a6-6bdb-11e8-8728-cee1298e900e.png">

adidalal commented:
> With higher security (SIP aka "Rootless") in 10.12-10.13, 3rd party drivers (kext) are placed in this temporary holding place until the user approves the use, at which time the driver is moved to /Library/Extensions.

So, this should not be an issue if installed on a fresh system, then approved.

It seems to occur when an app is installed, and then uninstalled without approving kext install.

MindTooth commented:
When referencing the kext without path, it does not remove anything.  `brew cask zap airparrot` states that it unloads the kext, but the four `kext`s are still present.  They are also present after a reboot follow the zapping ⚡️   

I'll removing the `StagedExtensions` lines.

MindTooth commented:
Just wanted to clarify; the reason I even made this PR, is because I wanted to remove left-over files, and remove it from `Sound Outputs`.  The current state of the PR does nothing towards that, so it has in a sense lost its purpose.

vitorgalvao commented:
What’s the status on this? Anything still needs to be done?

MindTooth commented:
Not sure, as it currently stands, it doesn't solve my issue.
If I'm not to delete all (staged), I still have AirParrot under Sound (Input/Output).

Been waiting for a response on the correct course forward.

adidalal commented:
@MindTooth Does this help?
http://help.airsquirrels.com/support/solutions/articles/1000185639-how-do-i-uninstall-airparrot-2-from-my-mac-

> To do this, press Shift+Command+G to bring up the Go-To-Folder option. In that search bar, type in /System/Library/Extensions and remove the "APExtFramebuffer.kext" and “AirParrotDriver.kext”. Move those files to the trash. 

> Once you do that, press the Shift+Command+G again and this time type in /Library/Extensions (they are different locations on a Mac). Again, remove the "APExtFramebuffer.kext" and “AirParrotDriver.kext” file by moving them to the trash.

MindTooth commented:
Updated the cask.

Read the docs one more time, and did a `eval (brew --repository)/Library/Taps/Homebrew/homebrew-cask/developer/bin/list_loaded_kext_ids` and found an additional kext.  Not sure if that one is the same as the other ones.  Anyways, the device is now removed from Sound AFAIK.

```sh
➜ eval (brew --repository)/Library/Taps/Homebrew/homebrew-cask/developer/bin/list_loaded_kext_ids
com.malwarebytes.mbam.rtprotection
com.squirrels.driver.AirParrotSpeakers
com.squirrels.airparrot.framebuffer
org.pqrs.driver.Karabiner.VirtualHIDDevice.v060800
org.virtualbox.kext.VBoxDrv
org.virtualbox.kext.VBoxUSB
org.virtualbox.kext.VBoxNetFlt
org.virtualbox.kext.VBoxNetAdp
com.github.kbfuse.filesystems.kbfuse
```
_(Obvously you learn the hard way not to remove a file called framebuffer.)_

vitorgalvao commented on Casks/airparrot.rb:
```diff
@@ -8,4 +8,15 @@
+                    '/System/Library/Extensions/AirParrotDriver.kext',
+                    '/System/Library/Extensions/APExtFramebuffer.kext',
+                  ],
+            quit: 'com.squirrels.AirParrot-2'
```
Does `quit: 'com.squirrels.AirParrot-*'` work? If so, that’s preferable. If not, `quit: "com.squirrels.AirParrot-#{version.major}"`.

MindTooth commented:
Worked with `"com.squirrels.AirParrot-#{version.major}"`.  Updated the cask.
